lldb/tools/debugserver/source/RNBRemote.cpp

```diff
@@ -2842,31 +2850,32 @@ rnb_err_t RNBRemote::SendStopReplyPacketForThread(nub_thread_t tid) {
     if (g_num_reg_entries == 0)
       InitializeRegisters();
 
-    if (g_reg_entries != NULL) {
-      auto interesting_regset = [](int regset) -> bool {
-#if defined(__arm64__) || defined(__aarch64__)
-        // GPRs and exception registers, helpful for debugging
-        // from packet logs.
-        return regset == 1 || regset == 3;
-#else
-        return regset == 1;
-#endif
-      };
-
-      DNBRegisterValue reg_value;
-      for (uint32_t reg = 0; reg < g_num_reg_entries; reg++) {
-        // Expedite all registers in the first register set that aren't
-        // contained in other registers
-        if (interesting_regset(g_reg_entries[reg].nub_info.set) &&
-            g_reg_entries[reg].nub_info.value_regs == NULL) {
-          if (!DNBThreadGetRegisterValueByID(
-                  pid, tid, g_reg_entries[reg].nub_info.set,
-                  g_reg_entries[reg].nub_info.reg, &reg_value))
-            continue;
-
-          debugserver_regnum_with_fixed_width_hex_register_value(
-              ostrm, pid, tid, &g_reg_entries[reg], &reg_value);
-        }
+    nub_size_t num_reg_sets = 0;
+    const DNBRegisterSetInfo *reg_sets = DNBGetRegisterSetInfo(&num_reg_sets);
+
+    DNBRegisterValue reg_value;
+    for (uint32_t reg = 0; reg < g_num_reg_entries; reg++) {
+      int regset = g_reg_entries[reg].nub_info.set;
+      bool include_reg = false;
+      // Expedite interesting register sets, all registers not
```

Comment:
Does this imply you are expediting svcr and tpidr2 as well?

I don't know debugserver but for Linux, only svl (svg) is needed in this scenario. Doesn't hurt to include more though.

Reply:
And by expedited I mean sent with every stop notification. If this code here is just what's sent with a general register read then it makes more sense to me.

Reply:
Yes, I figured as long as we were fetching ARM_SME_STATE register flavor to get svl, I might as well send svcr and tpidr2 as well, even if they're not being used today.

Reply:
You got me wondering how Linux tells whether it's in streaming mode or not. We rely on the fact that `vg` is the current vector length regardless of mode. So to update z0-31, we always use `vg`, and for `za` we use `svg` if we have it.

I think we would provide `vg` on a system with only streaming SVE, but I haven't tested it.

Given that, I think you expediting `svcr` is very sensible. If you have to deal with a bug along those lines, you won't have to guess what the mode is.